src/main.rs

```diff
@@ -606,8 +608,16 @@ async fn manifest(registry: Registry, info: web::Path<FetchInfo>) -> Result<Whar
             log::info(&format!("serving manifest from path: {:?}", &fq));
             match fs::read_to_string(&fq) {
                 Ok(manifest_str) => {
-                    let manifest = WharfixManifest::new(manifest_str, fq);
                     registry.blob_discovery(&path.join("blobs"));
+                    path.file_name().and_then(|path_base_name| {
+                        registry.store_blob(BlobInfo{
+                            name: path_base_name.to_str().unwrap().to_owned(),
+                            content_type: CONTENT_TYPE_MANIFEST.to_owned(),
+                            path: fq.clone()
+                        });
+                        Some("")
```

Comment:
Why `and_then` and `Some("")` over some fitting unwrap function? Feels smelly.

Reply:
`map(...)` it should have been, obviously.  :) Pushed a fixup now